sql/src/main/java/io/crate/analyze/MetaDataToASTNodeResolver.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.analyze;
+
+import com.google.common.collect.ImmutableMap;
+import io.crate.metadata.*;
+import io.crate.metadata.table.AbstractTableInfo;
+import io.crate.sql.tree.*;
+import io.crate.types.*;
+import org.elasticsearch.common.Nullable;
+
+import java.util.*;
+
+public class MetaDataToASTNodeResolver {
+
+    public static CreateTable resolveCreateTable(AbstractTableInfo info) {
```

Comment:
I think you could require a DocTableInfo here since the Analyzer casts to DocTableInfo anyway. That way the indexColumns can probably also be removed from the TableInfo interface and just need to be present in the DocTableInfo